src/registrar/management/commands/populate_domain_updated_federal_agency.py

```diff
@@ -0,0 +1,120 @@
+their new updated names ie (U.S. Peace Corps to Peace Corps)
+within Domain Information and Domain Requests
+"""
+
+import logging
+
+from django.core.management import BaseCommand
+from registrar.models import DomainInformation, DomainRequest, FederalAgency
+from registrar.management.commands.utility.terminal_helper import ScriptDataHelper
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Transfers Domain Request and Domain Information federal agency field from string to FederalAgency object"
+
+    # Deprecated federal agency names mapped to designated replacements {old_value, new value}
+    rename_deprecated_federal_agency = {
+        "Appraisal Subcommittee": "Appraisal Subcommittee of the Federal Financial Institutions Examination Council",
+        "Barry Goldwater Scholarship and Excellence in Education Program": "Barry Goldwater Scholarship and Excellence in Education Foundation",  # noqa
+        "Federal Reserve System": "Federal Reserve Board of Governors",
+        "Harry S Truman Scholarship Foundation": "Harry S. Truman Scholarship Foundation",
+        "Japan-US Friendship Commission": "Japan-U.S. Friendship Commission",
+        "Japan-United States Friendship Commission": "Japan-U.S. Friendship Commission",
+        "John F. Kennedy Center for Performing Arts": "John F. Kennedy Center for the Performing Arts",
+        "Occupational Safety & Health Review Commission": "Occupational Safety and Health Review Commission",
+        "Corporation for National & Community Service": "Corporation for National and Community Service",
+        "Export/Import Bank of the U.S.": "Export-Import Bank of the United States",
+        "Medical Payment Advisory Commission": "Medicare Payment Advisory Commission",
+        "U.S. Peace Corps": "Peace Corps",
+        "Chemical Safety Board": "U.S. Chemical Safety Board",
+        "Nuclear Waste Technical Review Board": "U.S. Nuclear Waste Technical Review Board",
+        "State, Local, and Tribal Government": "Non-Federal Agency",
+    }
+
+    def handle(self, **options):
+        """
+        Renames the Federal Agency to the correct new naming
+        for both Domain Information and Domain Requests objects.
+
+        NOTE: If it's NULL for a domain request, we skip it as
+        a user most likely hasn't gotten to it yet.
+        """
+        logger.info("Transferring federal agencies to FederalAgency object")
+        # DomainInformation object we populate with updated_federal_agency which are then bulk updated
+        domain_infos_to_update = []
+        domain_requests_to_update = []
+        # Domain Requests with null federal_agency that are not populated with updated_federal_agency
+        domain_requests_skipped = []
+        domain_infos_with_errors = []
+        domain_requests_with_errors = []
+
+        domain_infos = DomainInformation.objects.all()
+        domain_requests = DomainRequest.objects.all()
```

Comment:
Agreed Zander's suggestion is less expensive time wise, but for a single use script I'm not to worried about it. Did you test this on my sandbox or another sandbox with migration data? @therealslimhsiehdy and @erinysong This would indicate if the current process is really slow or totally fine for running on staging/stable